### PR TITLE
[com_menus] menus acl: don't allow to delete if user doesn't have permission to delete in that menu type

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
#### Summary of Changes

Some ACL in the menus list view aren't working quite right:
- If the user can't delete a particular menu type or is menu items it shouldn't be allowed to delete the menu type.
#### Testing Instructions
1. Create a user groups "testgroup" (child of "Administrator" user group)
2. Go to Menus -> Manager, create a "menu-dont-delete" menu type and set the "testgroup" group to have no permission (Denied) to "Delete"
3. Create a user "testuser" and add it only to "testgroup"
4. Open a new private browser window and login to backend with "testuser".
5. Go to menu Menus -> Manage
6. Try to delete the "menu-dont-delete". The menu is deleted!
   
   It shouldn't since you removed that permission in "2."
7. Apply patch. Repeat step 2. and 6. all good, a error message will appear.
   You can also try with several menus at the same time.

@bembelimen @infograf768 please test

### CODE REVIEW

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 72054a0664f572daa6edde2dd27afdd6b6138928<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

**Comment by Cybernary** [comment]
I have tested this item :white_check_mark: successfully on 72054a0664f572daa6edde2dd27afdd6b6138928<br /><br />Tested on 21-05-2016 and worked fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

File: administrator/components/com_menus/controllers/menus.php [hunk 1]
```diff
@@ -60,30 +60,45 @@ public function delete()
...
+
+			if (count($cids) > 0)
 			{
-				$this->setMessage(JText::plural('COM_MENUS_N_MENUS_DELETED', count($cid)));
+				// Get the model.
+				$model = $this->getModel();
+
+				// Make sure the item ids are integers
+				jimport('joomla.utilities.arrayhelper');
+				JArrayHelper::toInteger($cids);
```
Comment (roland-d) [inline administrator/components/com_menus/controllers/menus.php]: We should use Joomla\Utilities\ArrayHelper here as JArrayHelper is deprecated. Thanks.

**Comment by roland-d** [comment]
Taking RTC off for now until updated.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

**Comment by infograf768** [comment]
@roland-d
Can you merge after changing
```
				// Make sure the item ids are integers
				//jimport('joomla.utilities.arrayhelper');  // DELETE
				//JArrayHelper::toInteger($cids);  // DELETE
				$cids = Joomla\Utilities\ArrayHelper::toInteger($cids);  // ADD
```
as @andrepereiradasilva Is unavailable for now.

**Comment by andrepereiradasilva** [comment]
i will do it

**Comment by andrepereiradasilva** [comment]
done

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 88e40ed34c96c8235fd6d42d4f5f96d9df13bfa1<br /><br />One more tester.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

**Comment by infograf768** [comment]
@Cybernary 
We need another test here.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 88e40ed34c96c8235fd6d42d4f5f96d9df13bfa1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10547">issues.joomla.org/joomla-cms/10547</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**